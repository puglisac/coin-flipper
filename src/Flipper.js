import React, {useState} from 'react';
import Coin from './Coin';
import "./Flipper.css";


const Flipper= () => {
    const[heads, changeHeads] = useState(0);
    const[tails, changeTails] = useState(0);
    const [isHeads, flipCoin] =useState(null);
    function flip(){
        const random = Math.floor(Math.random()*2+1);
        if(random === 2){
            flipCoin(true);
            changeHeads(heads+1);
        }else {
            flipCoin(false)
            changeTails(tails+1);
        }
    }
    return (
    <div className="Flipper">
        <h1>Flip a Coin!!</h1>
    <div>
        {isHeads != null ? <Coin heads={isHeads}/> : null}
    </div>
    <p>Heads: {heads} Tails: {tails}</p>
    <button onClick={flip}>Flip Coin</button>
    </div>
    );
}

export default Flipper;