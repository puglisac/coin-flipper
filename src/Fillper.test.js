import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Flipper from './Flipper';

it("renders without crashing", function() {
  render(<Flipper />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Flipper />);
  expect(asFragment()).toMatchSnapshot();
});

it("coin is not displayed on page load", function(){
  const {queryByAltText}=render(<Flipper/>);
  expect(queryByAltText("coin-img")).not.toBeInTheDocument();
});

it("adds coin image on click and adds to either heads or tails", function(){
  const {queryByAltText, getByText}=render(<Flipper/>);
  const flip=getByText("Flip Coin");
  fireEvent.click(flip);
  expect(queryByAltText("coin-img")).toBeInTheDocument();
  expect(getByText("Heads", {exact:false})).toHaveTextContent(1);
});
