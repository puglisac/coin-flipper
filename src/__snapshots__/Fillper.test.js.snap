// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Flipper"
  >
    <h1>
      Flip a Coin!!
    </h1>
    <div />
    <p>
      Heads: 0 Tails: 0
    </p>
    <button>
      Flip Coin
    </button>
  </div>
</DocumentFragment>
`;
